tests/e2e/macvtap_test.go

```diff
@@ -0,0 +1,301 @@
+// Copyright 2020 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tests_test
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"strconv"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	MacvtapResource = "macvtap.network.kubevirt.io"
+	NetworkResource = "k8s.v1.cni.cncf.io/networks"
+	NetworkStatus = "k8s.v1.cni.cncf.io/networks-status"
+)
+
+const Timeout = 1 * time.Minute
+
+type ReportedNetwork struct {
+	Name      string            `json:"name"`
+	Interface string            `json:"interface"`
+	Mac       string            `json:"mac"`
+	Dns       map[string]string `json:"dns"`
+}
+
+var _ = Describe("macvtap-cni", func() {
+
+	lowerDevice := "eth0"
+	namespace := "default"
+
+	Describe("macvtap-cni infrastructure", func() {
+		Context("WHEN make cluster-sync is executed", func() {
+			It("THEN macvtap-cni daemonset is running in each k8s node", func() {
+				daemonSetNamePrefix := "macvtap-cni"
+				pods, _ := clientset.CoreV1().Pods(namespace).List(metav1.ListOptions{})
+				nodes, _ := clientset.CoreV1().Nodes().List(metav1.ListOptions{})
+				Expect(len(pods.Items)).Should(BeNumerically(">", 0))
+				Expect(len(nodes.Items)).Should(BeNumerically(">", 0))
+
+				networkStatus := FilterPods(pods.Items, func(pod v1.Pod) bool {
+					return strings.HasPrefix(pod.Name, daemonSetNamePrefix)
+				})
+				Expect(len(networkStatus)).To(Equal(len(nodes.Items)))
+			})
+
+			It("THEN a macvtap custom network resource is exposed", func() {
+				quantity := 50
+
+				expectedResourceName := v1.ResourceName(getMacvtapResourceName(lowerDevice))
+				waitForResourceAvailability(Timeout, getMacvtapResourceName(lowerDevice), metav1.ListOptions{})
```

Comment:
This is ok, but just so you know about another tool from Gomega:

You can implement your own `Eventually` https://github.com/nmstate/kubernetes-nmstate/blob/master/test/e2e/conditions.go#L58

Then it could look as following:

```
for node in nodes:
    nodeResourceCapacityEventually(reourceName).Should(Equal(expectedQuantity))
```

Reply:
Duly noted. One more trick for my bag of tricks.